### PR TITLE
Global java arguments

### PR DESCRIPTION
**Pull request by Ladvace**

### CODE REVIEW

File: app/actions/settings.js [hunk 1]
```diff
@@ -148,3 +149,13 @@ export function applyTheme(theme) {
     }
   };
 }
+
+export function Arg(args) {
```
Comment (blarfoon) [inline app/actions/settings.js]: Functions names should be camel-cased

File: app/components/InstanceManagerModal/Settings/Settings.js [hunk 2]
```diff
@@ -14,145 +14,162 @@ import { history } from '../../../store/configureStore';
...
 let watcher = null;
 
-class Instances extends Component<Props> {
-  props: Props;
+function Instances(props) {
+  const [instanceConfig, setInstanceConfig] = useState(null);
+  const [checkingForge, setCheckingForge] = useState(true);
+  const [unMounting, setUnMounting] = useState(false);
+  const [globalArg, setglobalArg] = useState();
+  const [OverArgs, setOverArgs] = useState(true);
```
Comment (blarfoon) [inline app/components/InstanceManagerModal/Settings/Settings.js]: Variables names should be camel-cased
Variable name is too generic

File: app/components/InstanceManagerModal/Settings/Settings.js [hunk 3]
```diff
@@ -14,145 +14,162 @@ import { history } from '../../../store/configureStore';
...
-    super(props);
-    this.state = {
-      instanceConfig: null,
-      checkingForge: true,
-      unMounting: false
-    };
-  }
-
-  componentDidMount = async () => {
+  async function ReadConfig(){
```
Comment (blarfoon) [inline app/components/InstanceManagerModal/Settings/Settings.js]: functions names should be camel-cased

File: app/components/InstanceManagerModal/Settings/Settings.js [hunk 4]
```diff
@@ -14,145 +14,162 @@ import { history } from '../../../store/configureStore';
...
 
-  render() {
-    const { getFieldDecorator } = this.props.form;
-    return (
-      <div>
-        <h2>Edit Instance Settings</h2>
-        <Form layout="inline" onSubmit={this.handleSubmit}>
-          <div>
-            <div
+  function OverrideArgs() {
```
Comment (blarfoon) [inline app/components/InstanceManagerModal/Settings/Settings.js]: Functions names should be camel-cased

File: app/components/InstanceManagerModal/Settings/Settings.js [hunk 5]
```diff
@@ -14,145 +14,162 @@ import { history } from '../../../store/configureStore';
...
+    setOverArgs(!OverArgs);
+    console.log(OverArgs);
+  }
+
+  function inputFunc(e) {
```
Comment (blarfoon) [inline app/components/InstanceManagerModal/Settings/Settings.js]: Function name is too generic

File: app/components/InstanceManagerModal/Settings/Settings.scss [hunk 6]
```diff
@@ -0,0 +1,12 @@
+@import '../../../style/theme/colors.scss';
+
+.JavaArginput {
```
Comment (blarfoon) [inline app/components/InstanceManagerModal/Settings/Settings.scss]: Css classes should be camel-cased

File: app/components/Settings/components/JavaManager/JavaArguments.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from 'react';
+import { Icon, Slider, Tooltip, Input, Button, message } from 'antd';
+import os from 'os';
+import fs from 'fs';
+import path from 'path';
+import store from '../../../../localStore';
+import { promisify } from 'util';
+import { connect } from 'react-redux';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import Constants from '../../../../constants';
```
Comment (blarfoon) [inline app/components/Settings/components/JavaManager/JavaArguments.js]: use ES6 destructuring

File: app/components/Settings/components/JavaManager/JavaArguments.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+import { Arg } from '../../../../actions/settings';
```
Comment (blarfoon) [inline app/components/Settings/components/JavaManager/JavaArguments.js]: Functions names should be camel-cased

File: app/components/Settings/components/JavaManager/JavaArguments.js [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+      : '';
+
+
+  let defaultARGS = `-Dfml.ignorePatchDiscrepancies=true -Dfml.ignoreInvalidMinecraftCertificates=true ${dosName}${
+    os.platform() === WINDOWS
+      ? '-XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump'
+      : ''
+    }-Xms256m -Xmx${props.ram}m`;
+
+  async function readJArgFile() {
```
Comment (blarfoon) [inline app/components/Settings/components/JavaManager/JavaArguments.js]: Wrong function name semantic

File: app/components/Settings/components/JavaManager/JavaArguments.js [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+    if (store.has('settings.java.javaArg')) {
+      const JavaArgumentsStoreSec = store.get('settings.java.javaArg');
```
Comment (blarfoon) [inline app/components/Settings/components/JavaManager/JavaArguments.js]: Variables names should be camel-cased

File: app/components/Settings/components/JavaManager/JavaArguments.js [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+      setglobalArg(JavaArgumentsStoreSec);
+      props.Arg(JavaArgumentsStoreSec);
+    } else {
+      const JavaArgumentsStore = store.set('settings.java.javaArg', defaultARGS);
```
Comment (blarfoon) [inline app/components/Settings/components/JavaManager/JavaArguments.js]: Variables names should be camel-cased

File: app/components/Settings/components/JavaManager/JavaArguments.js [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+      const JavaArgumentsStoreTh = store.get('settings.java.javaArg');
```
Comment (blarfoon) [inline app/components/Settings/components/JavaManager/JavaArguments.js]: Variables names should be camel-cased
Name too generic for a variable / difficult to understand

File: app/components/Settings/components/JavaManager/JavaArguments.js [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+  async function submit() {
+    props.Arg(globalArg);
+    if (globalArg) {
+      store.set('settings.java.javaArg', globalArg);
+    } else message.error("enter valid arguments");
+  }
+
+  function reset() {
+    const JavaArgumentsStore = store.set('settings.java.javaArg', defaultARGS);
```
Comment (blarfoon) [inline app/components/Settings/components/JavaManager/JavaArguments.js]: Variables names should be camel-cased

File: app/components/Settings/components/JavaManager/JavaArguments.js [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+    const JavaArgumentsStoreTh = store.get('settings.java.javaArg');
```
Comment (blarfoon) [inline app/components/Settings/components/JavaManager/JavaArguments.js]: Variables names should be camel-cased
Name too generic for a variable / difficult to understand

File: app/utils/MCLaunchCommand.js [hunk 15]
```diff
@@ -3,11 +3,13 @@ import fs from 'fs';
 import path from 'path';
 import { promisify } from 'util';
 import log from 'electron-log';
+import { connect } from 'react-redux';
+import javaArg from '../components/Settings/components/JavaManager/JavaArguments';
```
Comment (blarfoon) [inline app/utils/MCLaunchCommand.js]: Why importing a component into a utility function not directly related to react?

**Review by blarfoon — CHANGES_REQUESTED** [review]
Review code changes

File: app/components/InstanceManagerModal/Settings/Settings.js (hunk 2 above)
Comment (Ladvace) [reply app/components/InstanceManagerModal/Settings/Settings.js]: only the one on the 24 and 35 lines are mine

**Pull request closed without merging**